json/json.go

```diff
@@ -161,6 +161,23 @@ func AppendEscape(b []byte, s string, flags AppendFlags) []byte {
 	return b
 }
 
+// Unescape is a convenience helper to unescape a JSON value.
+// For more control over the unescape behavior and
+// to write to a pre-allocated buffer, use AppendUnescape.
+func Unescape(s []byte) []byte {
+	b := make([]byte, 0, len(s))
+	return AppendUnescape(b, s, ParseFlags(0))
+}
+
+// AppendUnescape appends s to b with the string unescaped as a JSON value.
+// This will include the starting and ending quote characters, and the
+// appropriate characters will be escaped correctly as if JSON decoded.
+func AppendUnescape(b []byte, s []byte, flags ParseFlags) []byte {
```

Comment:
Might be a nice to document the behavior if `b` does not have the capacity to store the decoded bytes.